### PR TITLE
[None][fix] Clean up Triton related tests

### PR DESCRIPTION
**Pull request by Tabrizian**
## Description

Cleans up the Triton-server integration tests and migrates QA coverage to L0 post-merge:

- **Remove broken/deprecated tests**: drops the multimodal (LLaVA, LLaVA-OneVision, MLLaMA, BLIP2, Qwen2-VL, Pixtral), speculative-decoding-BLS, disaggregated-serving-BLS, Medusa, and unit-test-wrapper Triton tests from `test_triton.py` / `test_triton_llm.py`, along with their entries in the test manifests and waives.
- **Remove dead helper code**: following the test removals, deletes the now-unused multimodal engine builders (`prepare_blip2_opt_engine`, `prepare_llava_engine`, `prepare_llava_onevision_engine`, `prepare_mllama_engine`, `prepare_mistral3_pixtral_engine`), the `prepare_multimodal_model_repo` / `prepare_disaggregated_serving_model_repo` / `modify_disaggregated_serving_config_pbtxt` helpers, the multimodal & `disaggregated_serving_bls` branches inside `modify_ib_config_pbtxt`, and the multimodal example-root / model-root fixtures. All removed symbols were verified to have no remaining references.
- **Restore `test_benchmark_core_model` dependencies**: the cleanup had accidentally removed `MODEL_CONFIGS`, the `model_setup` fixture, and `LATENCY_THRESHOLDS`, which `test_benchmark_core_model` still references — this caused a `NameError` at collection time. These are restored.
- **Migrate QA coverage to L0 post-merge**: promotes all still-valid QA Triton cases from `tests/integration/test_lists/qa/llm_triton_integration.txt` into the `l0_a30` `post_merge` / `triton` section (505 cases not already present), since the QA list is being deprecated. Hardware-gated cases (8-GPU 175b/70b, 80 GB return-logits) are gated by their own `skip_less_device` / `skip_less_device_memory` markers, consistent with existing cases in that section. The multi-node case is excluded (needs dedicated multi-node orchestration).

## Test Coverage

This PR only touches the Triton integration tests and their manifests; it adds no product code. Coverage is validated by the standard CI test-list validation (collection + manifest consistency) and by the migrated `l0_a30` `post_merge` / `triton` cases.

## PR Checklist

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

**Review by StanleySun639 — APPROVED** [review]

**Review by pcastonguay — APPROVED** [review]

File: tests/integration/defs/triton_server/test_triton_llm.py
Comment (pcastonguay) [inline tests/integration/defs/triton_server/test_triton_llm.py:2093]: Looks like tests are failing because `MODEL_CONFIGS` no longer exist.

**Comment by StanleySun639** [comment]
@Tabrizian TRTLLM is migrating QA role to L0 post-merge, could you please pick necessary test cases from [QA test list](https://github.com/NVIDIA/TensorRT-LLM/blob/main/tests/integration/test_lists/qa/llm_triton_integration.txt) into L0 post-merge coverage? Eventually QA test list will be depricated.  cc @schetlur-nv for awareness.

File: tests/integration/defs/triton_server/test_triton_llm.py [hunk 1]
```diff
@@ -2223,738 +2029,12 @@ def test_llama_v3_speculative_decoding_bls(
...
-        ENABLE_CONTEXT_FMHA_FP32_ACC=ENABLE_CONTEXT_FMHA_FP32_ACC,
-    )
-    # WAR for https://nvbugspro.nvidia.com/bug/4742149
-    gpu_name = query_gpu_name()
-    if "NVIDIA H20" == gpu_name:
-        check_call("pip3 install -U nvidia-cublas-cu12", shell=True)
+    # WAR for https://nvbugspro.nvidia.com/bug/4742149
+    gpu_name = query_gpu_name()
+    if "NVIDIA H20" == gpu_name:
+        check_call("pip3 install -U nvidia-cublas-cu12", shell=True)
```
[the remark this answers is not in the document]
Comment (Tabrizian) [reply tests/integration/defs/triton_server/test_triton_llm.py:1661]: This `pip3 install -U nvidia-cublas-cu12` workaround is pre-existing code (it dates back to PR #3549 and lives in `test_gpt_next_ptuning_ifb`, just shifted by surrounding test removals); it is not introduced or modified by this cleanup PR. Reworking it into the runner image / a setup fixture is a reasonable improvement but is out of scope for a test-cleanup PR (one concern per PR), so I'm leaving it unchanged here.

**Comment by Tabrizian** [comment]
Closing, since this was already completed here: https://github.com/NVIDIA/TensorRT-LLM/pull/15814#issuecomment-4872010162
